### PR TITLE
Bugfix 10343 missing validations via api

### PR DESCRIPTION
**Pull request by andrei7bal**
#10343 Add new validations for required fields when adding data from API
- add new validations for Sample and for TravelEntry
- use @NotEmpty annotation instead of @Requried (it's gonna be deprecated)

### CODE REVIEW

**Review by JonasCir — CHANGES_REQUESTED** [review]
Hey,
1. `NotEmpty` makes only sense on Strings, not other Objects. Therefore, use `NotNull`
2. I made a mistake: On Strings (if any use `NotBlank`, NOT `NotEmpty`
3. CI has a test failure

**Comment by andrei7bal** [comment]
Hello @JonasCir 
I've commited the changes. Please note that for the @Required annotation refactoring, there will be a new ticket (as I was suggested), so for this Pull Request only PointOfEntryDetails (TravelEntryDto) and SampleMaterialText (SampleDto) are treated.
Thanks for looking into this.

File: sormas-api/src/main/java/de/symeda/sormas/api/i18n/Validations.java [hunk 1]
```diff
@@ -241,12 +241,14 @@ public interface Validations {
 	String validPerson = "validPerson";
 	String validPhoneNumber = "validPhoneNumber";
 	String validPointOfEntry = "validPointOfEntry";
+	String validPointOfEntryDetails = "validPointOfEntryDetails";
```
Comment (JonasCir) [inline sormas-api/src/main/java/de/symeda/sormas/api/i18n/Validations.java:244]: remove this

File: sormas-api/src/main/java/de/symeda/sormas/api/i18n/Validations.java [hunk 2]
```diff
@@ -241,12 +241,14 @@ public interface Validations {
...
 	String validRegion = "validRegion";
 	String validReportDateTime = "validReportDateTime";
 	String validReportingUser = "validReportingUser";
 	String validResponsibleDistrict = "validResponsibleDistrict";
 	String validResponsibleRegion = "validResponsibleRegion";
 	String validSample = "validSample";
+	String validSampleMaterialText = "validSampleMaterialText";
```
Comment (JonasCir) [inline sormas-api/src/main/java/de/symeda/sormas/api/i18n/Validations.java:251]: remove this

File: sormas-api/src/main/java/de/symeda/sormas/api/sample/SampleDto.java [hunk 3]
```diff
@@ -108,6 +109,7 @@ public class SampleDto extends SormasToSormasShareableDto {
 	private SampleMaterial sampleMaterial;
 	@SensitiveData
 	@Size(max = FieldConstraints.CHARACTER_LIMIT_DEFAULT, message = Validations.textTooLong)
+	@NotBlank(message = Validations.validSampleMaterialText)
```
Comment (JonasCir) [inline sormas-api/src/main/java/de/symeda/sormas/api/sample/SampleDto.java:112]: should be `@NotBlank(message = Validations.requiredField)`

File: sormas-api/src/main/java/de/symeda/sormas/api/travelentry/TravelEntryDto.java [hunk 4]
```diff
@@ -104,6 +105,7 @@ public class TravelEntryDto extends PseudonymizableDto {
 	private PointOfEntryReferenceDto pointOfEntry;
 	@SensitiveData(mandatoryField = true)
 	@Size(max = FieldConstraints.CHARACTER_LIMIT_TEXT, message = Validations.textTooLong)
+	@NotBlank(message = Validations.validPointOfEntryDetails)
```
Comment (JonasCir) [inline sormas-api/src/main/java/de/symeda/sormas/api/travelentry/TravelEntryDto.java:108]: should be `@NotBlank(message = Validations.requiredField)`

**Comment by JonasCir** [comment]
hey, take a look at the issue again: The second scenario is not fixed.

**Comment by JonasCir** [comment]
tests are failing now

**Pull request merged**